<a href="https://colab.research.google.com/github/llpk79/DS-Unit-1-Sprint-4-Linear-Algebra/blob/master/Paul_K_Stats_review.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [5]:
import pandas as pd
url='https://github.com/pierretd/data-dump/raw/master/before-hours/Real_Estate_Valuation.csv'
df = pd.read_csv(url)

df.isna().sum()

No           0
Date         0
Age          0
Metro        0
Stores       0
Latitude     0
Longitude    0
Price        0
dtype: int64

In [6]:
df.describe()

No         Date         Age        Metro      Stores  \
count  414.000000   414.000000  414.000000   414.000000  414.000000   
mean   207.500000  2013.148971   17.712560  1083.885689    4.094203   
std    119.655756     0.281967   11.392485  1262.109595    2.945562   
min      1.000000  2012.667000    0.000000    23.382840    0.000000   
25%    104.250000  2012.917000    9.025000   289.324800    1.000000   
50%    207.500000  2013.167000   16.100000   492.231300    4.000000   
75%    310.750000  2013.417000   28.150000  1454.279000    6.000000   
max    414.000000  2013.583000   43.800000  6488.021000   10.000000   

         Latitude   Longitude       Price  
count  414.000000  414.000000  414.000000  
mean    24.969030  121.533361   37.980193  
std      0.012410    0.015347   13.606488  
min     24.932070  121.473530    7.600000  
25%     24.963000  121.528085   27.700000  
50%     24.971100  121.538630   38.450000  
75%     24.977455  121.543305   46.600000  
max     25.014590  121.566270  117.500000

The minimum age is 0.

The maximum value of stores is 10

The median longitude is 121.538630

The standard deviation of stores is 2.945562

In [7]:
df['Latitude'].skew()

-0.4385984450391146

In [9]:
df['Price'].kurtosis()

2.1790970477396163

In [10]:
corr_matrix = df.corr()
corr_matrix

No      Date       Age     Metro    Stores  Latitude  \
No         1.000000 -0.048658 -0.032808 -0.013573 -0.012699 -0.010110   
Date      -0.048658  1.000000  0.017549  0.060880  0.009635  0.035058   
Age       -0.032808  0.017549  1.000000  0.025622  0.049593  0.054420   
Metro     -0.013573  0.060880  0.025622  1.000000 -0.602519 -0.591067   
Stores    -0.012699  0.009635  0.049593 -0.602519  1.000000  0.444143   
Latitude  -0.010110  0.035058  0.054420 -0.591067  0.444143  1.000000   
Longitude -0.011059 -0.041082 -0.048520 -0.806317  0.449099  0.412924   
Price     -0.028587  0.087491 -0.210567 -0.673613  0.571005  0.546307   

           Longitude     Price  
No         -0.011059 -0.028587  
Date       -0.041082  0.087491  
Age        -0.048520 -0.210567  
Metro      -0.806317 -0.673613  
Stores      0.449099  0.571005  
Latitude    0.412924  0.546307  
Longitude   1.000000  0.523287  
Price       0.523287  1.000000

In [11]:
cov_matrix = df.cov()
cov_matrix

No       Date         Age         Metro       Stores  \
No         14317.500000  -1.641671  -44.723245 -2.049851e+03    -4.475787   
Date          -1.641671   0.079506    0.056372  2.166557e+01     0.008003   
Age          -44.723245   0.056372  129.788704  3.684082e+02     1.664189   
Metro      -2049.850694  21.665566  368.408249  1.592921e+06 -2239.938319   
Stores        -4.475787   0.008003    1.664189 -2.239938e+03     8.676334   
Latitude      -0.015012   0.000123    0.007694 -9.257892e+00     0.016236   
Longitude     -0.020309  -0.000178   -0.008483 -1.561822e+01     0.020302   
Price        -46.542615   0.335665  -32.640356 -1.156787e+04    22.885163   

           Latitude  Longitude         Price  
No        -0.015012  -0.020309    -46.542615  
Date       0.000123  -0.000178      0.335665  
Age        0.007694  -0.008483    -32.640356  
Metro     -9.257892 -15.618216 -11567.871845  
Stores     0.016236   0.020302     22.885163  
Latitude   0.000154   0.000079      0.092249  
Longitude  0.000079   0.000236      0.109273  
Price      0.092249   0.109273    185.136507